File: angular-web-project/src/app/model/payment-method.ts
import { Payment } from "./payment";
import { User } from "./user";

export type payment_type = 'TARJETA' | 'PAYPAL' | 'EFECTIVO' | string;

export interface PaymentMethod {
  method_id: number;
  user_id: number;
  type: payment_type;            // TARJETA | PAYPAL | EFECTIVO
  last4?: string | null;
  holder_name?: string | null;
  billing_address?: string | null;
  // Rel
  user?: User;
}

File: springboot-backend/hotelapi/src/main/java/com/runasagrada/hotelapi/model/PaymentMethod.java
package com.runasagrada.hotelapi.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class PaymentMethod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("method_id") 
    private Integer paymentMethodId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User userId;

    private String type;

    @JsonProperty("last4")
    private String lastfour;

    @JsonProperty("holder_name")
    private String holderName;

    @JsonProperty("billing_address")
    private String billingAddress;
}
